selftests/unit/test_utils_service.py

```diff
@@ -84,22 +90,28 @@ def setUp(self):
 
     def test_all_commands(self):
         command_name = "service"
+        # Test all commands except "set_target" which is tested elsewhere
         for cmd, _ in ((c, r) for (c, r) in
-                       self.service_command_generator.commands if
-                       c not in ["list", "set_target", "reset_failed", "mask",
-                                 "unmask"]):
+                       self.service_command_generator.commands
+                       if c != 'set_target'):
             ret = getattr(
                 self.service_command_generator, cmd)(self.service_name)
-            if cmd == "is_enabled":
-                command_name = "chkconfig"
-                cmd = ""
-            elif cmd == 'enable':
-                command_name = "chkconfig"
-                cmd = "on"
-            elif cmd == 'disable':
-                command_name = "chkconfig"
-                cmd = "off"
-            self.assertEqual(ret, [command_name, self.service_name, cmd])
+            if cmd in ['set_target', 'reset_failed', 'mask', 'unmask']:
+                exp = ['true']
+            elif cmd == 'list':
+                exp = ['chkconfig', '--list']
+            else:
+                if cmd == "is_enabled":
+                    command_name = "chkconfig"
+                    cmd = ""
+                elif cmd == 'enable':
+                    command_name = "chkconfig"
+                    cmd = "on"
+                elif cmd == 'disable':
+                    command_name = "chkconfig"
+                    cmd = "off"
+                exp = [command_name, self.service_name, cmd]
+            self.assertEqual(ret, exp, (ret, exp, cmd))
```

Comment:
Why did you set custom message `(ret, exp, cmd)` in `assertEqual()`?  For me implemented standard message output is better to read with hint on possible error spot:
See for yourself:
```
AssertionError: (['chkconfig', 'fake_service', 'on'], ['chkconig', 'fake_service', 'on'], 'on')
```
vs. implemented:
```
AssertionError: Lists differ: ['chkconfig', 'fake_service', ... != ['chkconig', 'fake_service', '...

First differing element 0:
chkconfig
chkconig

- ['chkconfig', 'fake_service', 'on']
?         -

+ ['chkconig', 'fake_service', 'on']

```

Reply:
I changed it to see the "cmd", which is not listed on failure. Anyway I can remove that (it was useful when adding the exceptions)